src/window_draw_list.rs

```diff
@@ -0,0 +1,617 @@
+use sys;
+use sys::{ImDrawCornerFlags, ImDrawList, ImU32};
+
+use super::{ImVec2, ImVec4, Ui};
+
+use std::marker::PhantomData;
+
+/// Wrap `ImU32` (a type typically used by ImGui to store packed colors)
+/// This type is used to represent the color of drawing primitives in ImGui's
+/// custom drawing API.
+///
+/// The type implements `From<ImU32>`, `From<ImVec4>`, `From<[f32; 4]>`,
+/// `From<[f32; 3]>`, `From<(f32, f32, f32, f32)>` and `From<(f32, f32, f32)>`
+/// for convenience. If alpha is not provided, it is assumed to be 1.0 (255).
+#[derive(Copy, Clone)]
+pub struct ImColor(ImU32);
+
+impl From<ImColor> for ImU32 {
+    fn from(color: ImColor) -> Self { color.0 }
+}
+
+impl From<ImU32> for ImColor {
+    fn from(color: ImU32) -> Self { ImColor(color) }
+}
+
+impl From<ImVec4> for ImColor {
+    fn from(v: ImVec4) -> Self { ImColor(unsafe { sys::igColorConvertFloat4ToU32(v) }) }
+}
+
+impl From<[f32; 4]> for ImColor {
+    fn from(v: [f32; 4]) -> Self { ImColor(unsafe { sys::igColorConvertFloat4ToU32(v.into()) }) }
+}
+
+impl From<(f32, f32, f32, f32)> for ImColor {
+    fn from(v: (f32, f32, f32, f32)) -> Self {
+        ImColor(unsafe { sys::igColorConvertFloat4ToU32(v.into()) })
+    }
+}
+
+impl From<[f32; 3]> for ImColor {
+    fn from(v: [f32; 3]) -> Self { [v[0], v[1], v[2], 1.0].into() }
+}
+
+impl From<(f32, f32, f32)> for ImColor {
+    fn from(v: (f32, f32, f32)) -> Self { [v.0, v.1, v.2, 1.0].into() }
+}
+
+/// All types from which ImGui's custom draw API can be used implement this
+/// trait. This trait is internal to this library and implemented by
+/// `WindowDrawList` and `ChannelsSplit`.
+pub trait DrawAPI<'ui> {
+    /// Get draw_list object
+    fn draw_list(&self) -> *mut ImDrawList;
+}
+
+/// Object implementing the custom draw API.
+pub struct WindowDrawList<'ui> {
+    draw_list: *mut ImDrawList,
```

Comment:
Do you know how lib wrappers usually deal with this kind of issues? It seems difficult to enforce such check at compile time.

I propose to enforce the `*mut ImDrawList` borrow check at run time. using a `static` variable to store the state. This may not be a perfect solution, but I hope it is optimal for a rusty wrapper of imgui. I implement this idea in patch 586e102, please let me know your thoughts.